keps/sig-node/2625-cpumanager-policies-thread-placement/README.md

```diff
@@ -183,80 +199,28 @@ With `threads_per_cpu` is typical 2 on x86_64 with SMT enabled - but this number
-| Baremetal, 2-way SMT capable (x86_64), SMT enabled   | 2                          |
-| Baremetal, 2-way SMT capable, SMT disabled           | 1                          |
-
-SMT can usually be disabled via software (kernel parameter, firmware settings), and [other implementations than 2-way SMT are possible](https://en.wikipedia.org/wiki/Simultaneous_multithreading).
-Hence, the ratio between virtual and physical CPUs is not fixed and is not predictable, subject to changes in future.
-Furthermore, this new unit would make sense only for CPU resources, and not for all the other resources.
-
-### Future extension
+### Test Plan
 
-#### Allowing external agents to detect the cpumanager behaviour
+[ ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
 
-As soon as a behaviour-changing option is added, we should add an alias to `static`, which we proposed to name `configurable`.
+##### Prerequisite testing updates
 
-This is motivated by the fact some operators may make logic on the reported CPUManager policy, and not notice the extra flags that potentially change the behaviour of the static policy.
-An example of these aforementioned operators are software components who vendored older kubelet configuration API (predating this work).
-By adding the new policy alias, we make it possible for those operators to trivially detect the change, and to not make wrong assumptions.
+##### Unit tests
 
-#### Emulating Non-SMT on SMT systems
+- `k8s.io/kubernetes/pkg/kubelet/cm/cpumanager`:            `20250130` - 85.6% of statements
+- `k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state`:      `20250130` - 88.1% of statements
+- `k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/topology`:   `20250130` - 85.0% of statements
 
-We would like to mention a further extension of this work, which we are *not* proposing here.
+##### Integration tests
 
-A further subset of the latency sensitive class of workload we identified (CNF, HFT) benefits most of non-SMT system, delivering the best possible performance here.
-For these applications, just disabling SMT at machine level solves the need of the workload, but overall creates worse usage of hardware resources and poorer container density.
+- kubelet features don't have usually integration tests. We use a combination of unit tests and `e2e_node` tests.
 
-Another policy option, or a further refinement of `full-pcpus-only`, which enables non-SMT emulation on SMT-enabled system would allow to accommodate these needs, but this would cause even more significant resource accounting mismatches
-as described above. Furthermore, at the moment of writing we are still assessing how large is the set of the classes which benefit of these extra guarantees.
+##### e2e tests
 
-For all these reasons we postponed this work to a later date.
+TBD
```

Comment:
/hold

We should probably have this filled out before merging.

Reply:
/hold cancel

done